### PR TITLE
Implement email login and account creation

### LINKED ISSUE #40: Support Email Login
Support email login for customers. Customer can click on `Login with Mail` and they can then enter email address and their password. If the user hasn't created an account, then create one if the credentials. Furthermore, check if the email isn't already used for Google logins.

The following methods from the Android app should be implemented.

```Kotlin

    fun fetchRegisteredAuthenticationSourcesForEmail(mailAddress: String): Single<List<AuthenticationSource>>

    fun createAccountWithMail(mailAddress: String, password: String): Completable

    fun loginWithMail(mailAddress: String, password: String): Completable
```

### PR DESCRIPTION
**Pull request by lockieRichter**
Implement the login screen for email and account creation bottom sheet. UI still needs polishing, but this works functionally for now and the layout/styling can be tweaked later.

Also missing a bunch of translations, unfortunately I can't speak a lot of German so will need @shockbytes to do the translations in another change. It's probably worth adding something to the README about the process for translations for other developers (for example at my work we don't use Google to do translations, we always make sure to get a fluent speaker of the language to do it).

This is a pretty massive change, so I might have missed some stuff!

**Enter email**
image

**Invalid email**
image

image

**Email already in use**
image

**Existing user**
image

**New user**
image

**Invalid password**
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +81,12 @@ Developers should be aiming for tests to cover all the major functionality. We a
 generate mocks required for testing. To be able to run the tests, you'll need to run
 `flutter pub run build_runner build` from the repo root to generate the mocks required for the tests.
 
+### String translations
```
Comment (shockbytes) [inline README.md:84]: 👏

File: README.md [hunk 2]
```diff
@@ -81,6 +81,12 @@ Developers should be aiming for tests to cover all the major functionality. We a
...
+
+All static text strings should be in the translations files located at `/lib/l10n`.
```
Comment (shockbytes) [inline README.md:86]: I think the folder is `/I10n` not `/l10n`

File: lib/src/ui/login/login_page.dart [hunk 3]
```diff
@@ -128,13 +129,14 @@ class LoginPageState extends State<LoginPage> {
                         ),
                       ),
                       OutlinedButton(
-                        onPressed: () => _bloc.loginWithEmail(),
+                        // onPressed: () => _bloc.loginWithEmail(),
```
Comment (shockbytes) [inline lib/src/ui/login/login_page.dart]: Can be removed.

File: lib/src/ui/login/login_page.dart [hunk 4]
```diff
@@ -19,12 +21,12 @@ class LoginPage extends StatefulWidget {
 class LoginPageState extends State<LoginPage> {
   final LoginBloc _bloc = DependencyInjector.get<LoginBloc>();
   late StreamSubscription<LoginEvent> _loginSubscription;
-  late bool _isLoggingIn;
+  late bool _loading;
```
Comment (shockbytes) [inline lib/src/ui/login/login_page.dart]: Please use the `is` prefix => `_isLoading`

**Review by shockbytes — CHANGES_REQUESTED** [review]
Please update the german translations as well to this PR. I will then translate this directly on the PR.

**Comment by shockbytes** [comment]
@lockieRichter Please add the strings in the german file too, otherwise we might miss them. I think it's better to have all translations contained in such a PR.

File: README.md (hunk 2 above)
Comment (lockieRichter) [reply README.md:86]: The translations are actually in the folder `/l10n`, are you thinking of `i18n`?

File: lib/src/ui/login/login_page.dart (hunk 3 above)
Comment (lockieRichter) [reply lib/src/ui/login/login_page.dart]: Good catch

File: lib/src/ui/login/login_page.dart (hunk 4 above)
Comment (lockieRichter) [reply lib/src/ui/login/login_page.dart]: Good catch.

**Comment by lockieRichter** [comment]
@shockbytes can have a go at the translations, I don't speak much German unfortunately. Should I use Google translate or just put the translations in English?

File: README.md (hunk 2 above)
Comment (shockbytes) [reply README.md:86]: yeah, sorry. `l10n` is right. I was indeed thinking about `i18n`. :D

**Comment by shockbytes** [comment]
@lockieRichter I would suggest that you add the entries in the `app_de.arb` file too and use `"TODO"` placeholders for them. I will provide the german translation for it either by directly pushing a commit to the PR (as done now) or in one of my branches.

**Review by shockbytes — APPROVED** [review]

**Pull request merged**